react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -1116,8 +1115,19 @@ function _mapStateToProps(state, ownProps): Object {
     const { localFlipX } = state['features/base/settings'];
     const _isMobile = isMobileBrowser();
     const activeParticipants = getActiveParticipantsIds(state);
+    let layout = _currentLayout;
+
+    if (_currentLayout === LAYOUTS.STAGE_FILMSTRIP_VIEW) {
+        if (stageFilmstrip) {
+            layout = LAYOUTS.TILE_VIEW;
```

Comment:
This whole thing with rewriting the layout value and then using it seems very confusing to me. 
Lets just add `LAYOUTS.STAGE_FILMSTRIP_VIEW` to the appropriate places in the component. If it is not that straight forward  maybe you can have a functions that are checking everything necessary and returning a boolean.

Reply:
I see this layout is passed to other components as well and it is a long chain. Another option will be to have a notion of filmstrip type with new enumeration of values for tile view type, vertical view type and horizontal view type. You can then rename _currentLayout here and trough the chain to filmstripType and use it this way. I think this will look way clearer to me.

Right now the problem is that how we display the thumbnail and its parts doesn't depend on the layout itself since we have a stage filmstrip one which have 2 different types of filmstrip inside. It looks like this more related to the type of filmstrip that renders the thumbnail. WDYT?

Reply:
Yes, I agree this is kinda confusing. Considering this is only used for determining how to display the thumbnails I think it does make sense to introduce a Thumbnail type instead of a filmstrip type because for eg on stage filmstrip we want to display TILE type thumbnails